### ISSUE TITLE
option to control white space between tokens and indentation

### ISSUE THREAD

**Issue by millermedeiros** [estools/escodegen#4]
I got to this project while discussing ways to do advanced code formatting for JS ([source](https://twitter.com/#!/millermedeiros/status/178501265813291008))

It would be great if it had the same level of control as [FDT advanced code formatter](http://fdt.powerflasher.com/docs/Advanced_code_formatter_%28AS3_%2B_MXML%29) - options to define how white space is handled between each kind of token (line break, space, none), indentation, keep/remove empty lines, etc.

Basically transform _escodegen_ into a hardcore version of [JSBeautifier](http://jsbeautifier.org/).

**Comment by asdf23** [comment]
I would like to have the code formatted with leading commas. Such as:
var a = ({
     a: "Prefixed with tab then space"
    ,b: "Prefixed with tab then comma"
    ,c: "Prefixed with tab then comma"
});
